tests/components/bbox/test_device_tracker.py

```diff
@@ -0,0 +1,177 @@
+    with patch("homeassistant.components.bbox.device_tracker.pybbox") as mock_pybbox:
+        mock_box = MagicMock()
+        mock_pybbox.Bbox.return_value = mock_box
+        yield mock_box
+
+
+@pytest.fixture
+def device_tracker_config():
+    """Return device tracker configuration."""
+    return {DEVICE_TRACKER_DOMAIN: {"host": "192.168.1.254"}}
+
+
+async def test_get_scanner_success(
+    hass: HomeAssistant, device_tracker_config: ConfigType, mock_bbox_api
+) -> None:
+    """Test successful scanner initialization."""
+    mock_bbox_api.get_all_connected_devices.return_value = [
+        {
+            "macaddress": "aa:bb:cc:dd:ee:ff",
+            "hostname": "test_device",
+            "ipaddress": "192.168.1.100",
+            "active": 1,
+        }
+    ]
+
+    scanner = get_scanner(hass, device_tracker_config)
+    assert scanner is not None
+    assert scanner.success_init is True
+
+
+async def test_get_scanner_failure(
+    hass: HomeAssistant, device_tracker_config: ConfigType, mock_bbox_api
+) -> None:
+    """Test scanner initialization failure."""
+    mock_bbox_api.get_all_connected_devices.side_effect = requests.exceptions.HTTPError(
+        "Connection failed"
+    )
+    with patch.object(BboxDeviceScanner, "_update_info", return_value=False):
+        scanner = get_scanner(hass, device_tracker_config)
+        assert scanner is None
+
+
+async def test_scan_devices(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test scanning for devices."""
+    with patch.object(BboxDeviceScanner, "_update_info") as mock_update:
+        # Create scanner without calling __init__ to avoid API calls
+        scanner = BboxDeviceScanner.__new__(BboxDeviceScanner)
+        scanner.host = device_tracker_config[DEVICE_TRACKER_DOMAIN]["host"]
+        scanner.success_init = True
+        scanner.last_results = [
+            Device("aa:bb:cc:dd:ee:ff", "test_device", "192.168.1.100", None)
+        ]
+
+        devices = scanner.scan_devices()
+        assert devices == ["aa:bb:cc:dd:ee:ff"]
+        mock_update.assert_called_once()
+
+
+async def test_get_device_name(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test getting device name by MAC address."""
+    # Create scanner without calling __init__ to avoid API calls
+    scanner = BboxDeviceScanner.__new__(BboxDeviceScanner)
+    scanner.host = device_tracker_config[DEVICE_TRACKER_DOMAIN]["host"]
+    scanner.success_init = True
+    scanner.last_results = [
+        Device("aa:bb:cc:dd:ee:ff", "test_device", "192.168.1.100", None),
+        Device("ff:ee:dd:cc:bb:aa", "another_device", "192.168.1.101", None),
+    ]
+
+    # Test existing device
+    name = scanner.get_device_name("aa:bb:cc:dd:ee:ff")
+    assert name == "test_device"
+
+    # Test non-existing device
+    name = scanner.get_device_name("11:22:33:44:55:66")
+    assert name is None
+
+
+async def test_update_info_success(
```

Comment:
Update_info is a private method. I don't think we need to tests for it. We should tests public methods like `get_device_name`, `scan_devices`, or '__init__`.